File: src/main/java/com/ject/studytrip/cleanup/application/executor/HardDeleteExecutor.java
package com.ject.studytrip.cleanup.application.executor;

import java.util.function.LongSupplier;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class HardDeleteExecutor {

    public long run(String phase, LongSupplier supplier) {
        try {
            long deleted = supplier.getAsLong();
            log.info("[HardDelete] phase={}, deleted={}", phase, deleted);
            return deleted;
        } catch (Exception e) {
            log.error("[HardDelete] phase={} failed: {}", phase, e.getMessage(), e);
            return -1L; // 실패 표식
        }
    }
}

File: src/main/kotlin/com/ject/studytrip/cleanup/application/executor/HardDeleteExecutor.kt
package com.ject.studytrip.cleanup.application.executor

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class HardDeleteExecutor {
    companion object {
        private val log = LoggerFactory.getLogger(HardDeleteExecutor::class.java)
    }

    fun run(
        phase: String,
        supplier: () -> Long,
    ): Long =
        try {
            val deleted = supplier()
            log.info("[HardDelete] phase={}, deleted={}", phase, deleted)
            deleted
        } catch (e: Exception) {
            log.error("[HardDelete] phase={} failed: {}", phase, e.message, e)
            -1L // 실패 표식
        }
}
